### PR TITLE
Fixes to CAR hitmap interpolation order and sky fraction calculation + homogenous scan strategy debug mode

### LINKED ISSUE #38: Healpix and CAR hitmaps are not equivalent
There are two bugs in the CAR implementation that cause the effective sky fraction for CAR to be different from that in the healpix implementation, thus causing them to pull different noise curves from the SO noise model. The bugs are:

1. The default sky fraction calculation assumes equal area pixels. For CAR, one can switch this to a calculation that sums over a map of the pixel areas in the unmasked region.
2. When converting an existing CAR hitmap to a different geometry, it uses enmap.project which by default uses order=3 interpolation. This should be switched to order=0 (which is what the original healpix -> CAR and a healpy.ud_grade operation does).

### PR DESCRIPTION
**Pull request by msyriac**
This fixes #38. It is ready to be reviewed and merged.

### CODE REVIEW

**Comment by msyriac** [comment]
Added a feature that allows for homogenous "scan strategy", fixing #40 .

File: mapsims/noise.py [hunk 1]
```diff
@@ -185,46 +193,65 @@ def update_telescope(self, telescope, scanning_strategy):
...
+                from pixell import enmap, wcsutils
+
+                hitmap = enmap.read_map(hitmap_filename)
+                if wcsutils.is_compatible(hitmap.wcs, self.wcs):
+                    hitmap = enmap.extract(hitmap, self.shape, self.wcs)
+                else:
+                    warnings.warn(
+                        "WCS of hitmap with nearest pixel-size is not compatible, so interpolating hitmap"
+                    )
+                    hitmap = enmap.project(hitmap, self.shape, self.wcs,order=0)
```
Comment (msyriac) [inline mapsims/noise.py]: Adding order=0 is a key fix of this PR.

File: mapsims/noise.py [hunk 2]
```diff
@@ -185,46 +193,65 @@ def update_telescope(self, telescope, scanning_strategy):
...
+
+            hitmap /= hitmap.max()
+            
+            # Discard pixels with very few hits that cause border effects
+            # hitmap[hitmap < 1e-3] = 0
+            if self.healpix:
+                self.sky_fraction[telescope] = (hitmap != 0).sum() / hitmap.size
+            else:
+                self.sky_fraction[telescope] = enmap.pixsizemap(self.shape,self.wcs)[hitmap!=0].sum() / 4. / np.pi
```
Comment (msyriac) [inline mapsims/noise.py]: This pixel area sum is another key fix of this PR.

File: mapsims/noise.py [hunk 3]
```diff
@@ -45,6 +45,7 @@ def __init__(
         SA_number_UHF=2,
         SA_one_over_f_mode="pessimistic",
         hitmap_version="v0.1",
+        fsky=None,
```
Comment (zonca) [inline mapsims/noise.py]: can you please rename variable to `sky_fraction`?

File: mapsims/noise.py [hunk 4]
```diff
@@ -85,7 +86,11 @@ def __init__(
             If True, reduce the hitcount by a factor of 0.85 to account for not-uniformity in the scanning
         scanning_strategy : str
             Choose between the available scanning strategy hitmaps "classical" or "opportunistic" or
-            path to a custom hitmap; it will be normalized, so absolute hitcount does not matter
+            path to a custom hitmap; it will be normalized, so absolute hitcount does not matter.
+            If None, or "homogenous" or "isotropic", then no hitmap scaling is applied and 
+            a homogenous noise field is generated on the full sky. In this case, the sky fraction fsky must be
+            specified, which will be used to populate the full sky with homogenous noise with noise power
+            equal to what SO would obtain if it observed only a partial sky of fraction fsky.
```
Comment (zonca) [inline mapsims/noise.py]: I have modified this section also in my other standing pull request. We will have to merge the 2 changesets somehow

**Comment by zonca** [comment]
@msyriac this pull request needs to be updated to the newest changes in `mapsims`, do you think you'll be able to work on it?

**Comment by msyriac** [comment]
I see that the new hitmap versions are different for each channel, so they are loaded from disk when the simulation is requested. For CAR, an intermediate reprojection step might be needed, so for each simulation one has a disk i/o operation (+ a potential reprojection for CAR). To address this, I will cache the hitmaps for each channel if ok with @zonca .

**Comment by zonca** [comment]
@msyriac sure

**Comment by msyriac** [comment]
Running another change by @zonca :
I think it is safer to use tuple seeds,
e.g. something schematically like
seed = (self.seed,frequency_id,telescope_id,nsplits)
instead of constructing a single integer, which requires more thought to avoid seed collisions.

**Comment by zonca** [comment]
> Running another change by @zonca :
> I think it is safer to use tuple seeds,
> e.g. something schematically like
> seed = (self.seed,frequency_id,telescope_id,nsplits)
> instead of constructing a single integer, which requires more thought to avoid seed collisions.

nice! I didn't know you could do that, it's way cleaner

**Comment by msyriac** [comment]
FYI, I'm retiring this pull request and starting a new one since it seems easier to merge with master after starting from scratch. Let's leave this open (so I can use it for reference) until the new one is merged.

**Comment by zonca** [comment]
ok, if rebasing is too messy, also consider doing a merge of master into your branch

**Comment by msyriac** [comment]
The changes attempted here are now in #44 , so I'm closing this.

**Pull request closed without merging**